src/v3/src/util/idxUtils.ts

```diff
@@ -229,3 +238,90 @@ export const convertIdxMessageToWidgetMessage = (
     message: getMessage(message),
   };
 });
+
+export const convertIdxInputsToRegistrationSchema = (
+  inputs: Input[],
+): RegistrationElementSchema[] => (
+  inputs.reduce((acc: Input[], input: Input) => {
+    const flattenedInputs = flattenInputs(input);
+    return [...acc, ...flattenedInputs];
+  }, []).map((flattendInput: Input) => ({
+    ...flattendInput,
+    'label-top': true,
+    'data-se': flattendInput.name,
+    wide: true,
+  } as RegistrationElementSchema))
+);
```

Comment:
The schema array that is sent to the customer in the callback is a flattened version of the `Input` array provided to use from IDX. So this function just flattens the objects and adds default values for those extra properties (which is what v2 sets them as too).